src/test/java/org/jenkinsci/plugins/mesos/ConnectionTest.java

```diff
@@ -33,12 +39,28 @@ public void startAgent(TestUtils.JenkinsRule j)
     String mesosUrl = mesosCluster.getMesosUrl();
     MesosApi api = new MesosApi(mesosUrl, "example", "MesosTest");
 
-    MesosSlave agent = api.enqueueAgent().toCompletableFuture().get();
+    MesosCloud cloud = Mockito.mock(MesosCloud.class);
```

Comment:
The connection test is actually an integration test and not a unit test. What do you think about creating a `MesosApiTest` and `MesosCloudTest` for unit testing?

I'm still on the fence for mocking.

Reply:
ok added kill integration test here will move the mocks out to the unit tests

Reply:
ok added new integration tests